### PR TITLE
Adjustment of the voting scheme and updated viral taxonomy

### PR DESCRIPTION
**Pull request by hoelzer**
Recent changes introduced by @guille0387 in the context of solving some pending manuscript revision.

We both tested the branch on example data, and I already open the PR while still some things need to be solved:

To run the code, a new taxonomy sqlite file is needed, which should be added to the auto-download procedure of the pipeline before merging this PR. e.g. available here:

* https://u.pcloud.link/publink/show?code=XZ9eYEVZLuaHtlP4ryRyYBJRQ3bm78myPDwX

Also, this file is needed to be updated (as unzipped version):

* [additional_data_vpHMMs_v3.tsv.zip](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/11599995/additional_data_vpHMMs_v3.tsv.zip)

Then, assuming that the new `additional_data_vpHMMs_v3.tsv` is in the `nextflow-autodownload-databases/models` folder, once can run this branch via

```bash
nextflow run virify.nf --fasta kleiner_virome_2015.fasta --output results/kleiner_2023 --ncbi nov_2022_taxdb.sqlite --virome -profile local,docker
```

Further, we discovered that the container for chromomap is not available: `quay.io/microbiome-informatics/r_chromomap:0.2`

@mberacochea can you please add it to your registry?

### CODE REVIEW

**Comment by mberacochea** [comment]
> Recent changes introduced by @guille0387 in the context of solving some pending manuscript revision.
> 
> We both tested the branch on example data, and I already open the PR while still some things need to be solved:
> 
> To run the code, a new taxonomy sqlite file is needed, which should be added to the auto-download procedure of the pipeline before merging this PR. e.g. available here:
> 
>     * https://u.pcloud.link/publink/show?code=XZ9eYEVZLuaHtlP4ryRyYBJRQ3bm78myPDwX
> 
> 
> Also, this file is needed to be updated (as unzipped version):
> 
>     * [additional_data_vpHMMs_v3.tsv.zip](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/11599995/additional_data_vpHMMs_v3.tsv.zip)
> 
> 
> Then, assuming that the new `additional_data_vpHMMs_v3.tsv` is in the `nextflow-autodownload-databases/models` folder, once can run this branch via
> 
> ```shell
> nextflow run virify.nf --fasta kleiner_virome_2015.fasta --output results/kleiner_2023 --ncbi nov_2022_taxdb.sqlite --virome -profile local,docker
> ```
> 
> Further, we discovered that the container for chromomap is not available: `quay.io/microbiome-informatics/r_chromomap:0.2`
> 
> @mberacochea can you please add it to your registry?

Hey @hoelzer and @guille0387,

This is great, Guille sent me the new version of "additional_data_vpHMMs_v3.tsv" a few ago .. I've been away for a bit, so just catching up.
A few things:
- [ ] We should rename additional_data_vpHMMs_v3.tsv -> additional_data_vpHMMs_v4.tsv (or similar). We should keep v3 of the file around, some people may need it in case they used the current release.
- [ ] I'll fix the container, strange that is missing.
- [ ] Sure, I'll update the .sql file too.
- [ ] We should rename `contig_taxonomic_assign_2.py` => `contig_taxonomic_assign.py`
- [ ] Add a small python unit test for that script, @guille0387 do you have a small toy example I can use for the test?

Please let me know if this make sense.

**Comment by hoelzer** [comment]
Hey @mberacochea !

I think v3 for the `additional_data_vpHMMs_v3.tsv` is fine, because the current version of the pipeline has only v1 and v2:

https://github.com/EBI-Metagenomics/emg-viral-pipeline/blob/master/nextflow/modules/metaGetDB.nf#L22

Besides, I agree with your ToDo list! 

Another thing: we only worked on the nextflow code to get things done for the paper revision. The changes then also need to be reflected in CWL.

thx!

**Comment by guille0387** [comment]
Hi @mberacochea,

I agree with what @hoelzer said in the previous comment. Also, here's a small file that you can use for the test.
[test_set.tsv.zip](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/11609928/test_set.tsv.zip)

**Comment by mberacochea** [comment]
> Hey @mberacochea !
> 
> I think v3 for the `additional_data_vpHMMs_v3.tsv` is fine, because the current version of the pipeline has only v1 and v2:
> 
> https://github.com/EBI-Metagenomics/emg-viral-pipeline/blob/master/nextflow/modules/metaGetDB.nf#L22
> 
> Besides, I agree with your ToDo list!
> 
> Another thing: we only worked on the nextflow code to get things done for the paper revision. The changes then also need to be reflected in CWL.
> 
> thx!

:man_facepalming: ... for some reason I was conviced the file we had was v3... anyway. Perfect then.

CWL... our team is moving to Nextflow, so unless CWL is on the paper I wouldn't spend time on it. We can just freeze on v1 for CWL.

**Comment by hoelzer** [comment]
> CWL... our team is moving to Nextflow, so unless CWL is on the paper I wouldn't spend time on it. We can just freeze on v1 for CWL.

Ah, good to know. Then we could maybe also remove the CWL part from the manuscript @guille0387 when this is anyway not longer supported.

**Comment by hoelzer** [comment]
Hey @mberacochea we are in the process of finalizing the new data and results for the paper revision. So far, it looks really good: we get more taxonomical annotations e.g. on the 243 TARA ocean MAGs, compared to the former pipeline version and backend taxonomy. 

It would be great if we can finalize this PR and make another release by the end of June to also put the new pipeline version that we're using now into the manuscript. What's the status regarding the ToDos you listed? Can we help w/ smt? thx!

**Comment by mberacochea** [comment]
> Hey @mberacochea we are in the process of finalizing the new data and results for the paper revision. So far, it looks really good: we get more taxonomical annotations e.g. on the 243 TARA ocean MAGs, compared to the former pipeline version and backend taxonomy.
> 
> It would be great if we can finalize this PR and make another release by the end of June to also put the new pipeline version that we're using now into the manuscript. What's the status regarding the ToDos you listed? Can we help w/ smt? thx!

That is great! Apologies for the delay on my side. I agree, we need this one merged and a new release of the pipeline ASAP. I'll try to make that happen as soon as I can. I don't think there is anything in particular I need help with. I'll keep you both posted.

**Comment by hoelzer** [comment]
Hey @mberacochea , @guille0387 and I are on finalizing the manuscript revision - any updates regarding the PR and open ToDos? thx a lot!

File: tests/contig_taxonomic_assign/expected_assignment_taxonomy.tsv
Comment (mberacochea) [inline tests/contig_taxonomic_assign/expected_assignment_taxonomy.tsv:1]: @guille0387 Can you check if this expected file is valid? I create it using the contig_assignment script, and it's used in the unit test.

**Comment by mberacochea** [comment]
> Hey @mberacochea , @guille0387 and I are on finalizing the manuscript revision - any updates regarding the PR and open ToDos? thx a lot!

Hey.. almost there. I just need to fix the container and upload the files.. and we should be done.

File: tests/contig_taxonomic_assign/expected_assignment_taxonomy.tsv
Comment (guille0387) [reply tests/contig_taxonomic_assign/expected_assignment_taxonomy.tsv:1]: Hi @mberacochea , yes the file looks alright and contains the expected output 👍🏻

**Comment by mberacochea** [comment]
Folks. I'm going to merge this one.. and will run some tests using DEV

**Comment by hoelzer** [comment]
> Folks. I'm going to merge this one.. and will run some tests using DEV

awesome! thx so much @mberacochea !

**Pull request merged**